### PR TITLE
Using tail recursion to collect Kafka client properties

### PR DESCRIPTION
**Pull request by arpanchaudhury**

### CODE REVIEW

File: core/src/main/scala/akka/kafka/internal/ConfigSettings.scala [hunk 1]
```diff
@@ -4,32 +4,31 @@
...
+
+import scala.collection.JavaConverters.asScalaSetConverter
 
 /**
  * INTERNAL API
  */
 private[kafka] object ConfigSettings {
 
   def parseKafkaClientsProperties(config: Config): Map[String, String] = {
-    def collectKeys(c: ConfigObject, prefix: String, keys: Set[String]): Set[String] = {
```
Comment (patriknw) [inline core/src/main/scala/akka/kafka/internal/ConfigSettings.scala]: Did you see the issue for real, or is this just a nice cleanup?
Wouldn't it have to be a really deep properties structure to be an issue.

Comment (arpanchaudhury) [inline core/src/main/scala/akka/kafka/internal/ConfigSettings.scala]: @patriknw I didn't faced any issue as such but considering the number of properties we can pass to the producers and consumers I refactored it as a nice to have. WDYT?

Comment (patriknw) [inline core/src/main/scala/akka/kafka/internal/ConfigSettings.scala]: I think it's good to improve it, thanks.

To clarify, for the archives, the old grows stack per prop level, not per property. E.g. aaa.bbb.ccc.ddd.eee.fff is 6 levels.

**Comment by arpanchaudhury** [comment]
@patriknw thank you for the explanation.  updated the PR.

**Comment by kciesielski** [comment]
LGTM, thanks @arpanchaudhury!

**Pull request merged**